### PR TITLE
tech note on DAMP/DRY

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: content/blog/2024-03-19-package-testing-principles/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+If the test were to fail, in the console we could run `devtools::load_all()` and run the code of the test, as `devtools::load_all()` loads testthat helper files so `basic_list()` would be available.
+
+
+## Conclusion
+
+The balance between DRY ("Don't repeat yourself") and DAMP ("Descriptive and meaningful phrases") is a trade-off.
+To keep the water analogy, we also don't want code whose effects leak unexpectedly.
+What we should strive for are self-contained test that we can understand and run without too much context.
+
```
Comment (steffilazerte) [inline content/blog/2024-03-19-package-testing-principles/index.Rmd:120]:
```suggestion

Another consideration we haven't covered here, are tests that require specific elements, such as environment variables or options. In those cases try using [withr](https://withr.r-lib.org/) functions such as `[withr::local_envvar()](https://withr.r-lib.org/reference/with_envvar.html)` in each test that requires it.

```

File: content/blog/2024-03-19-package-testing-principles/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+One powerful idea from the [book "Software Engineering at Google"](https://www.oreilly.com/library/view/software-engineering-at/9781492082781/) by Titus Winters, Tom Manshreck, Hyrum Wright, is that code can afford to be a bit less obvious because it has tests covering it, whereas test code isn't covered by test code.
+
+For elements such as environment variables and options, we would use calls such as `withr::local_envvar()` in each test.
```
Comment (steffilazerte) [inline content/blog/2024-03-19-package-testing-principles/index.Rmd:123]: These seemed a bit out of place here. What do you think of putting it above with a bit more tying in?
```suggestion
```

**Review by steffilazerte — COMMENTED** [review]
Very cool @maelle! I love these posts, I learn a lot!

I did find it a bit choppy at times, so I've added some connections and descriptions to things. I hope that doesn't make it too wordy, see what you think!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [ ] title is in Title Case
* [X] publication date is ok
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [ ] YAML subject tags are ok ("tech notes" for tech notes; "community" for non-staff non-editor)
* [X] YAML field 'preface' is present if necessary

**Comment by maelle** [comment]
Thank you @steffilazerte!! Your comments are very helpful as always.
